### PR TITLE
Create `dex.atomic_arbitrages`

### PR DESCRIPTION
**Pull request by hildobby**
# Thank you for contributing to Spellbook!
Please refer to the top of the `readme` in the root of Spellbook to learn how to contribute to Spellbook on DuneSQL.

### CODE REVIEW

**Comment by hildobby** [comment]
First time all tests pass after my first commit for a new spell

Screenshot 2023-10-20 at 02 53 57

now back to work

File: models/dex/dex_schema.yml [hunk 1]
```diff
@@ -426,7 +426,8 @@ models:
       - *token_bought_amount_raw
       - *token_sold_amount
       - *token_bought_amount
-      - name: amount_usd
+      - &amount_usd
+        name: amount_usd
         description: "USD value of the trade at time of execution"
```
Comment (Hosuke) [inline models/dex/dex_schema.yml:431]: https://github.com/duneanalytics/spellbook/blob/00a467a599cef7baa0bebd58428ced936bd1e7cd/models/dex/dex_schema.yml#L62
`amount_usd` defined at L62 before.

File: .github/workflows/dbt_slim_ci.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ concurrency:
 jobs:
   dbt-test:
     runs-on: [ self-hosted, linux, spellbook-trino-ci ]
-    timeout-minutes: 90
+    timeout-minutes: 300
```
Comment (jeff-dude) [inline .github/workflows/dbt_slim_ci.yml:20]: let's be sure to revert this

File: macros/models/_sector/dex/dex_atomic_arbitrages.sql [hunk 3]
```diff
@@ -0,0 +1,129 @@
+{% macro dex_atomic_arbitrages(blockchain, transactions) %}
+
+-- Step 1: Check that the transaction contains at least 2 trades
+WITH multi_trade_txs AS (
+    SELECT block_time, tx_hash
+    FROM {{ ref('dex_trades') }}
+    WHERE blockchain = '{{blockchain}}'
+    {% if is_incremental() %}
+    AND block_time >= date_trunc('day', now() - interval '7' day)
```
Comment (jeff-dude) [inline macros/models/_sector/dex/dex_atomic_arbitrages.sql:9]: similar to last PR, let's update all incremental filters

File: macros/models/_sector/dex/dex_atomic_arbitrages.sql [hunk 4]
```diff
@@ -0,0 +1,129 @@
+{% macro dex_atomic_arbitrages(blockchain, transactions) %}
+
+-- Step 1: Check that the transaction contains at least 2 trades
+WITH multi_trade_txs AS (
+    SELECT block_time, tx_hash
+    FROM {{ ref('dex_trades') }}
```
Comment (jeff-dude) [inline macros/models/_sector/dex/dex_atomic_arbitrages.sql:6]: we're reading from `dex.trades` multiple times throughout. i wonder if performance wise, it's better to have a simple CTE at the top where you read from `dex.trades`, filter on blockchain and incremental reads, only pull columns which you need downstream. then each subsequent CTE reads from that one and does each step along the way.

Comment (jeff-dude) [reply macros/models/_sector/dex/dex_atomic_arbitrages.sql:6]: do you mind taking note of current CI test runtimes with code as-is now, then push a commit to try that, and see if we see any significant gains?

File: macros/models/_sector/dex/dex_atomic_arbitrages.sql [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+, pst.token_bought_amount_raw
+, pst.amount_usd
+, pst.taker
+, pst.maker
+, pst.project_contract_address
+, pst.tx_from
+, pst.tx_to
+, txs.index AS tx_index
+FROM positive_sum_trades
+MATCH_RECOGNIZE (
```
Comment (jeff-dude) [inline macros/models/_sector/dex/dex_atomic_arbitrages.sql:113]: out of curiosity, are you implementing this logic based on [this PR](https://github.com/duneanalytics/spellbook/pull/4891/files#diff-ab8b7cfc938c5312fdc456e5d996285b0f5af0fca623216ba22b9df774dc977d)?

File: models/dex/chains/mev/arbitrages/dex_arbitrum_atomic_arbitrages.sql [hunk 6]
```diff
@@ -0,0 +1,15 @@
+{{ config(
+        schema = 'dex_arbitrum',
```
Comment (jeff-dude) [inline models/dex/chains/mev/arbitrages/dex_arbitrum_atomic_arbitrages.sql:2]: maybe we implement the same `blockchain` variable approach?

File: models/dex/chains/mev/arbitrages/dex_arbitrum_atomic_arbitrages.sql [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+        alias = 'atomic_arbitrages',
+        partition_by = ['block_month'],
+        materialized = 'incremental',
+        file_format = 'delta',
+        incremental_strategy = 'merge',
+        unique_key = ['blockchain','tx_hash', 'evt_index']
```
Comment (jeff-dude) [inline models/dex/chains/mev/arbitrages/dex_arbitrum_atomic_arbitrages.sql:8]: can we look to add that target incremental filter on these too, similar to the dex sandwiched?

**Review by jeff-dude — CHANGES_REQUESTED** [review]

File: macros/models/_sector/dex/dex_atomic_arbitrages.sql (hunk 5 above)
Comment (hildobby) [reply macros/models/_sector/dex/dex_atomic_arbitrages.sql:113]: I spoke with @kasiafi last week who guided me through this, and yes this PR served as a good guide for here

File: .github/workflows/dbt_slim_ci.yml (hunk 2 above)
Comment (hildobby) [reply .github/workflows/dbt_slim_ci.yml]: done

File: macros/models/_sector/dex/dex_atomic_arbitrages.sql (hunk 3 above)
Comment (hildobby) [reply macros/models/_sector/dex/dex_atomic_arbitrages.sql]: done

File: macros/models/_sector/dex/dex_atomic_arbitrages.sql (hunk 4 above)
Comment (jeff-dude) [reply macros/models/_sector/dex/dex_atomic_arbitrages.sql:27]: based on monitored run times, the previous setup was most efficient. plz revert back and we can continue 🙏

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**